### PR TITLE
Flow everywhere

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] Adds `@flow` everywhere
- [x] Fixes 1 bug!
- [x] Cleans up package.json and lets our child scripts be like the rest of our code (flow/es6)
- [x] Icons are sorted now (stop dumb diffs)

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -29,7 +29,7 @@
         "babel/func-params-comma-dangle": 0,
         "comma-dangle": [2, "always-multiline"],
         "filenames/match-regex": [2, "^[0-9a-z-.]+(\\.desktop|\\.native|\\.ios|\\.android)?$"],
-        "flowtype/require-valid-file-annotation": [2, "never"], // TODO always once we get 100% clean
+        "flowtype/require-valid-file-annotation": [2, "always"],
```
Comment (chrisnojima) [inline .eslintrc]: aww yisssss

File: desktop/app/store-helper.js [hunk 2]
```diff
@@ -46,7 +47,7 @@ export default function (mainWindow) {
 
     // Reverse so the indexes don't shift
     dead.reverse().forEach(idx => {
-      subscribeStoreSubscribers.splice(dead, 1)
+      subscribeStoreSubscribers.splice(idx, 1)
```
Comment (chrisnojima) [inline desktop/app/store-helper.js]: this was a real bug! dead tracker windows likely weren't cleaning up correctly

File: desktop/npm-helper.js [hunk 3]
```diff
@@ -38,11 +39,13 @@ function pad (s, num) {
   return s
 }
 
+const nodeCmd = 'babel-node --presets es2015,stage-2 --plugins transform-flow-strip-types'
```
Comment (chrisnojima) [inline desktop/npm-helper.js]: have our child scripts be able to run w/ es2015 and flow

File: desktop/npm-helper.js [hunk 4]
```diff
@@ -352,7 +357,7 @@ const iconMeta_ = {
 ${
   // eslint really doesn't understand embedded backticks
 /* eslint-disable */
-Object.keys(icons).map(name => {
+Object.keys(icons).sort().map(name => {
```
Comment (chrisnojima) [inline desktop/npm-helper.js]: sort our icons to reduce the amount of churn in this script

File: desktop/npm-vendor.js [hunk 5]
```diff
@@ -89,9 +96,9 @@ function updateVendored () {
   // force HTTPS for more efficient cloning
   vendorURL = vendorURL.replace(/^git@github.com:/, 'https://github.com/')
   var vendorCommit = spawn('git', ['rev-parse', 'HEAD'], {cwd: VENDOR_DIR, stdio: 'pipe'}).stdout.trim()
-  var packageInfo = JSON.parse(fs.readFileSync('./package.json'))
+  var packageInfo = JSON.parse(fs.readFileSync('./package.json', {encoding: 'utf8'}))
```
Comment (chrisnojima) [inline desktop/npm-vendor.js]: dunno how this worked w/o the encoding. it should send back a buffer

File: desktop/package.json [hunk 6]
```diff
@@ -4,31 +4,33 @@
...
-    "inject-sourcemaps-prod": "babel-node --presets es2015,stage-2 npm-helper.js inject-sourcemaps-prod",
-    "inject-code-prod": "babel-node --presets es2015,stage-2 npm-helper.js inject-code-prod",
-    "start-prod": "babel-node --presets es2015,stage-2 npm-helper.js start-prod",
-    "start-hot-debug": "babel-node --presets es2015,stage-2 npm-helper.js start-hot-debug",
-    "debug-main": "babel-node --presets es2015,stage-2 npm-helper.js debug-main",
-    "setup-debug-main": "babel-node --presets es2015,stage-2 npm-helper.js setup-debug-main",
-    "postinstall": "babel-node --presets es2015,stage-2 npm-helper.js postinstall",
-    "render-screenshots": "babel-node --presets es2015,stage-2 npm-helper.js render-screenshots",
-    "updated-fonts": "babel-node --presets es2015,stage-2 npm-helper.js updated-fonts",
+    "_node": "babel-node --presets es2015,stage-2 --plugins transform-flow-strip-types ",
```
Comment (chrisnojima) [inline desktop/package.json]: some helpers to reduce all the copy paste

File: shared/nav.android.js [hunk 7]
```diff
@@ -110,7 +113,7 @@ class Nav extends Component {
 
     const drawerContent = (
       <Box style={{flex: 1, backgroundColor: '#fff'}}>
-        <Text style={{margin: 10, fontSize: 15, textAlign: 'left'}}>I'm in the Drawer!</Text>
+        <Text type='Header' style={{margin: 10, fontSize: 15, textAlign: 'left'}}>I'm in the Drawer!</Text>
```
Comment (chrisnojima) [inline shared/nav.android.js]: no type= before???

File: shared/settings/about.js [hunk 8]
```diff
@@ -1,18 +0,0 @@
-import React, {Component} from 'react'
```
Comment (chrisnojima) [inline shared/settings/about.js]: just killing all this old non-used settings junk 👊

**Comment by keybase-ci-visdiff** [comment]
The commits 6371a058a8f2...e33c6309995b introduce visual changes on linux. <details><summary>:mag_right: 2 changed</summary>

 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475588004672-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Image-6371a058a8f2.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475588004672-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Image-e33c6309995b.png)  
   Icon-Icon_Image comparison
 * Changed: **Icon-Icon_Icon_Font** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475588004672-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Icon_Font-6371a058a8f2.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475588004672-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Icon_Font-e33c6309995b.png)  
   Icon-Icon_Icon_Font comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 6371a058a8f2...e33c6309995b introduce visual changes on linux. <details><summary>:mag_right: 2 changed</summary>

 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475591298049-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Image-6371a058a8f2.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475591298049-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Image-e33c6309995b.png)  
   Icon-Icon_Image comparison
 * Changed: **Icon-Icon_Icon_Font** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475591298049-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Icon_Font-6371a058a8f2.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475591298049-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Icon_Font-e33c6309995b.png)  
   Icon-Icon_Icon_Font comparison
</summary>

**Comment by chrisnojima** [comment]
@keybase/react-hackers please take a look

**Comment by keybase-ci-visdiff** [comment]
The commits f666fb7a535b...afe17cc9a22f introduce visual changes on linux. <details><summary>:mag_right: 2 changed</summary>

 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475618168565-f666fb7a535b-afe17cc9a22f-linux/Icon-Icon_Image-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475618168565-f666fb7a535b-afe17cc9a22f-linux/Icon-Icon_Image-afe17cc9a22f.png)  
   Icon-Icon_Image comparison
 * Changed: **Icon-Icon_Icon_Font** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475618168565-f666fb7a535b-afe17cc9a22f-linux/Icon-Icon_Icon_Font-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475618168565-f666fb7a535b-afe17cc9a22f-linux/Icon-Icon_Icon_Font-afe17cc9a22f.png)  
   Icon-Icon_Icon_Font comparison
</summary>

File: desktop/npm-vendor.js [hunk 9]
```diff
@@ -1,11 +1,16 @@
...
+// @flow
 
 var os = require('os')
 var path = require('path')
 var fs = require('fs')
 var spawnSync = require('child_process').spawnSync
 
-var VENDOR_DIR = process.env.KEYBASE_JS_VENDOR_DIR
+var VENDOR_DIR = process.env.KEYBASE_JS_VENDOR_DIR || ''
+if (!VENDOR_DIR) {
```
Comment (chromakode) [inline desktop/npm-vendor.js]: This duplicates a check near the bottom of the script. Should it be moved up?

File: desktop/npm-vendor.js [hunk 10]
```diff
@@ -89,9 +96,9 @@ function updateVendored () {
   // force HTTPS for more efficient cloning
   vendorURL = vendorURL.replace(/^git@github.com:/, 'https://github.com/')
   var vendorCommit = spawn('git', ['rev-parse', 'HEAD'], {cwd: VENDOR_DIR, stdio: 'pipe'}).stdout.trim()
-  var packageInfo = JSON.parse(fs.readFileSync('./package.json'))
+  var packageInfo = JSON.parse(fs.readFileSync('./package.json', {encoding: 'utf8'}))
   packageInfo.keybaseVendoredDependencies = `${vendorURL}#${vendorCommit}`
-  fs.writeFileSync('./package.json', JSON.stringify(packageInfo, 2, 2))
+  fs.writeFileSync('./package.json', JSON.stringify(packageInfo, null, 2))
```
Comment (chromakode) [inline desktop/npm-vendor.js]: Oops! Thanks `flow`.

File: desktop/package.json [hunk 11]
```diff
@@ -4,32 +4,35 @@
...
-    "setup-debug-main": "babel-node --presets es2015,stage-2 npm-helper.js setup-debug-main",
-    "postinstall": "babel-node --presets es2015,stage-2 npm-helper.js postinstall",
-    "render-screenshots": "babel-node --presets es2015,stage-2 npm-helper.js render-screenshots",
-    "local-visdiff": "babel-node --presets es2015,stage-2 npm-helper.js local-visdiff",
-    "updated-fonts": "babel-node --presets es2015,stage-2 npm-helper.js updated-fonts",
-    "wrap": "npm uninstall fsevents && npm prune && npm dedupe && npm shrinkwrap --dev",
-    "shrinkwrap-audit": "babel-node --presets es2015,stage-2 npm-shrinkwrap-audit",
-    "help": "babel-node --presets es2015,stage-2 npm-helper.js help",
-    "watch-test-file": "KEYBASE_RUN_MODE=devel babel-node --presets es2015,stage-2 npm-helper.js watch-test-file",
+    "_helper": "npm run _node -- npm-helper.js",
```
Comment (chromakode) [inline desktop/package.json]: So much nicer!

File: shared/profile/index.desktop.js [hunk 12]
```diff
@@ -1,15 +1,16 @@
 /* @flow */
```
Comment (chromakode) [inline shared/profile/index.desktop.js]: `// @flow`?

File: shared/settings/about.js (hunk 8 above)
Comment (chromakode) [inline shared/settings/about.js]: :scissors: :heart:

File: shared/profile/index.desktop.js (hunk 12 above)
Comment (chrisnojima) [inline shared/profile/index.desktop.js]: yeah, i went ahead and changed that everywhere

**Comment by chrisnojima** [comment]
@chromakode please take a look at the new commits. i added a couple of new rules so we have comma dangling like our regular js code, removed semicolons also. changed all /* style to // and addressed your PR feedback (also a master merge)
so close!

**Comment by chrisnojima** [comment]
also ticketed an option to force this /* vs // style. https://github.com/gajus/eslint-plugin-flowtype/issues/126

**Pull request merged**